Detectors/TRD/workflow/src/TRDGlobalTrackingSpec.cxx

```diff
@@ -683,19 +704,32 @@ bool TRDGlobalTracking::refitTPCTRDTrack(TrackTRD& trk, float timeTRD, o2::globa
 bool TRDGlobalTracking::refitTRDTrack(TrackTRD& trk, float& chi2, bool inwards)
 {
   auto propagator = o2::base::Propagator::Instance();
+
   int lyStart = inwards ? NLAYER - 1 : 0;
   int direction = inwards ? -1 : 1;
   int lyEnd = inwards ? -1 : NLAYER;
   o2::track::TrackParCov* trkParam = nullptr;
   o2::track::TrackLTIntegral* tofL = nullptr;
+  auto matCorr = o2::base::Propagator::MatCorrType(mRec->GetParam().rec.trd.matCorrType);
+
   if (inwards) {
     trkParam = &trk;
     tofL = &trk.getLTIntegralOut();
   } else {
     trkParam = &trk.getOuterParam();
     trkParam->setUserField(trk.getUserField()); // pileup timing info
+
+    const auto& trackTune = o2::globaltracking::TrackTuneParams::Instance();
+    if (trackTune.tpcCovOuterType != o2::globaltracking::TrackTuneParams::AddCovType::Disable) {
+      if (!propagator->PropagateToXBxByBz(*trkParam, o2::constants::geom::XTPCOuterRef, o2::base::Propagator::MAX_SIN_PHI, o2::base::Propagator::MAX_STEP, matCorr)) {
```

Comment:
Should this not only be done for the outwards refit and not for the inwards refit, as there we are starting at the outermost TRD point?
Since the covariance matrix is reset for the inwards fit in any case I think it does not matter, but this would just add unnecessary back and forth propagation, no?

Reply:
Hi @martenole 

> Should the covariance not be updated also for the TPC-TRD tracks at the inner TPC radius?

I am expanding the TPC cov matrix only before a fit/matching in the external detector, to fix the matching. The TPC cov. itself (w/o ITS update) is not used at the moment in the analysis (unless the V0 finder with TPC-only tracks is used, in which case the error will propagate to V0 error), but in any case, I  would prefer to expand track errors right before using them for the refit: the only purpose of this operation is to improve the matching eff. Otherwise, I would need to expand bare TPC track errors at their reconstruction level.

>  And just out of curiosity: why is there no correction added for Z in https://github.com/AliceO2Group/O2DPG/pull/869 ?

Z is not used in the matching and I could not calibrate the sigma to add from the difference of TPC and ITS tracks at the matching point, since in Z TPC track is simply moved to match the ITS. Still, I could add the same sigma for the symmetry reasons as for Y.

> Should this not only be done for the outwards refit and not for the inwards refit, as there we are starting at the outermost TRD point? Since the covariance matrix is reset for the inwards fit in any case I think it does not matter, but this would just add unnecessary back and forth propagation, no?

For the inwad refit w/o ITS update see above. For the outward: the point is that the track refitted in the TPC has distortion on both edges and these distortions are not accounted for by the track cov. matrix. So, (for the TRD refit in this case) I manipulate the TPC cov. matrix at the outer boundary of the TPC in order to not affect the TRD outer param by TPC distortions. Same I do to minimize their effect on the [matching](https://github.com/shahor02/AliceO2/blob/3a23e785dbd7cf523eec2c3dbe7292725f5037f8/Detectors/TRD/workflow/src/TRDGlobalTrackingSpec.cxx#L331). That said, at the moment the updateCov at TPC outer radius is a placeholder: I did not activate `trackTune.tpcCovOuterType`.

Reply:
Ah, sorry, I misread the changes in github and see it better in the code directly now. In case of the TRD refit it is only done before the outwards refit and not before the inwards fit.
Thank you for the clarifications